modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/domain/DefaultMuleDomain.java

```diff
@@ -232,9 +244,21 @@ public void dispose() {
     if (logger.isInfoEnabled()) {
       logger.info(miniSplash(format("Disposing domain '%s'", getArtifactName())));
     }
+
+    ClassLoader originalClassloader = currentThread().getContextClassLoader();
+
+    if (originalClassloader != deploymentClassLoader.getClassLoader()) {
```

Comment:
you may use `withContextClassLoader` here too, extracting the below code to a lambda